src/Admin/Controller/SendTestMailController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Admin\Controller;
+
+use Flarum\Foundation\Application;
+use Flarum\Settings\TemporarySettingsRepository;
+use Flarum\User\AssertPermissionTrait;
+use Illuminate\Contracts\Validation\Factory;
+use Illuminate\Support\Arr;
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Swift_Mailer;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class SendTestMailController implements RequestHandlerInterface
+{
+    use AssertPermissionTrait;
+
+    protected $app;
+
+    protected $translator;
+
+    public function __construct(Application $app, TranslatorInterface $translator)
+    {
+        $this->app = $app;
+        $this->translator = $translator;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
```

Comment:
We could probably drastically simplify this method with the following approach:
- Assume settings are saved.
- *Only* the test recipient is then needed in the payload (depending on the discussion about using the admin's email). It's not a settings field like all the others, it's only needed for this little transaction.
- You can now just inject the driver, ask it whether it `canSend()` for error handling, and get the transport / mailer from it.

What do you think?

Reply:
While that would be cleaner code, I don't think we should be saving potentially invalid settings. To be honest, I would prefer if the "Save Settings" button was disabled until a successful test email has been sent, that way it's impossible to add in incorrect config. Thoughts?